README.md

```diff
@@ -19,102 +22,125 @@ npm i --save object-rewrite
 
 ## Getting Started
 
-Modifies the data object in place. If you need to create a copy consider using [_.deepClone()](https://lodash.com/docs/#cloneDeep).
-
-<!-- eslint-disable-next-line import/no-unresolved -->
+<!-- eslint-disable import/no-unresolved, import/no-extraneous-dependencies -->
 ```js
-const objectRewrite = require('object-rewrite');
-
-const data = [{
-  guid: 'aad8b948-a3de-4bff-a50f-3d59e9510aa9',
-  count: 3,
-  active: 'yes',
-  tags: [{ id: 1 }, { id: 2 }, { id: 3 }]
-}, {
-  guid: '4409fb72-36e3-4385-b3da-b4944d028dcb',
-  count: 4,
-  active: 'yes',
-  tags: [{ id: 2 }, { id: 3 }, { id: 4 }]
-}, {
-  guid: '96067a3c-caa2-4018-bcec-6969a874dad9',
-  count: 5,
-  active: 'no',
-  tags: [{ id: 3 }, { id: 4 }, { id: 5 }]
-}];
-
-const rewriter = objectRewrite({
-  inject: {
-    '': (key, value, parents) => ({ countNext: value.count + 1 })
-  },
-  overwrite: {
-    active: (key, value) => value === 'yes'
-  },
-  filter: {
-    '': (key, value, parents) => value.active === true,
-    tags: (key, value, parents) => value.id === 4
-  },
-  retain: ['count', 'countNext', 'active', 'tags.id']
+const {
+  injectPlugin,
+  filterPlugin,
+  sortPlugin,
+  rewriter
+} = require('object-rewrite');
+
+const queryDataStore = (fields) => { /* ... */ };
+
+const inject = injectPlugin({
+  target: 'idNeg',
+  requires: ['id'],
+  fn: ({ value }) => -value.id
+});
+const filter = filterPlugin({
+  target: '*',
+  requires: ['idNeg'],
+  fn: ({ value }) => [-2, -1].includes(value.idNeg)
 });
+const sort = sortPlugin({
+  target: '*',
+  requires: ['idNeg'],
+  fn: ({ value }) => value.idNeg
+});
+const rew = rewriter({ '': [inject, filter, sort] });
+
+const desiredFields = ['id'];
+const rewInstance = rew(desiredFields);
+
+const data = queryDataStore(rewInstance.toRequest);
+// data => [{ id: 0 }, { id: 1 }, { id: 2 }]
 
-rewriter(data);
-
-// => data is now modified
-/*
-[{
-  "count": 3,
-  "countNext": 4,
-  "active": true,
-  "tags": []
-}, {
-  "count": 4,
-  "countNext": 5,
-  "active": true,
-  "tags": [{"id": 4}]
-}]
-*/
+rewInstance.rewrite(data);
+// data => [{ id: 2 }, { id: 1 }]
 ```
 
-The empty needle `""` matches top level object(s).  
+Please see the tests for more in-depth examples on how to use this library.
+
+## Plugins
+
+There are three types of plugins `INJECT`, `FILTER` and `SORT`.
+
+All plugins require a `target` of type `String`, a `required` of type `Array` and `fn` of type `Functions`.
+
+- `target`: specifies the relative field or object this plugin acts on
+- `required`: specifies the relative required fields. Will influence `toRequest`.
+- `fn`: result of this function is used by the plugin. Signature is `fn({ key, value, parents, context })`.
```

Comment:
Have a look at how I've done the types [here](https://github.com/loopmediagroup/data-sdk/blob/master/README.md#create)
```suggestion
- `fn` _Function_: result of this function is used by the plugin. Signature is `fn({ key, value, parents, context })`.
```